overlord/stateengine.go

```diff
@@ -117,6 +122,35 @@ func (se *StateEngine) StartUp() error {
 	return nil
 }
 
+func (se *StateEngine) maybeRestartHeldTasks() error {
+	s := se.state
+	s.Lock()
+	defer s.Unlock()
+
+	for _, ch := range s.Changes() {
+		var chBootId string
+		if err := ch.Get("boot-id", &chBootId); err != nil {
+			continue
+		}
+
+		currentBootID, err := osutil.BootID()
+		if err != nil {
+			return err
+		}
+		if currentBootID != chBootId {
+			logger.Debugf("restart already happened, moving forward change %s", ch.ID())
+			for _, t := range ch.Tasks() {
+				if t.Status() != state.HoldStatus {
```

Comment:
this is too imprecise, there might be other tasks that are in HoldStatus but for other reasons, it seems we need to mark also the task a bit further

Reply:
Done now